### DISCUSSION TITLE
Need help for spanning time range and interval selection

### DISCUSSION THREAD

**Discussion by rollysys** [vega/altair#3311] [Q&A]
The data is a range of linux scheduler slice, which is like:
|ts|ts_end|cpu|cpu2|
|0|10|1|1|
|5|20|2|2|
I have a altair chart below.
```python
interval = alt.selection_interval(bind="scales",encodings=['x'])
d=(alt.Chart(dfsched.reset_index())
    .mark_bar()
    .properties(
    width=2000,
    height=200)
    .encode(alt.X("ts:Q").scale(zero=False), 
            alt.X2("ts_end:Q"), 
            y="cpu:Q", 
            y2="cpu2:Q", 
            tooltip=["ts:Q", "name:O", "dur:Q"],
            color=alt.Color('name:N', scale=alt.Scale(scheme='turbo')).legend(None)
            )
    .add_params(interval)
)
```

And I have another chart to calculate the histogram in the extent of selection internal(zoom range acatually)
But altair will calculate the 'ts>interval.min()' by default, that will cause the data incorrect because many slices is dropped which are'ts<interval.min() and ts_end<interval.max()"
so could I use span operation on ts/ts_end and interval?
I find span is supported in vega expression but it's diffcult for me to use it.
And this is an example from perfetto:
![image](https://github.com/altair-viz/altair/assets/72813698/8ffd4be8-6d1c-43db-869d-28de8c7b3c2d)

**Comment by mattijn** [comment]
Thank you for starting this discussion. I try to understand the issue you are having, but unfortunately the current information given is not enough. 

It's already great that you provide your Altair specification as text, but to provide meaningful feedback it is much appreciated if you could try to elaborate, preferably with a minimal reproducible example (including data), that can be copied, that showcases the problem you are having. Thank you!

**Reply by rollysys** [reply]
Thanks and I made an example in vega-lite editor below(already desensitization)
[gist](https://gist.github.com/rollysys/44462224788eacee45d7ebfd3b45470d)
Let me explain the interactive behavior and my problem:
The main data is in chart below, which is some thing like a gannt chart, the bar represents some cpu event on a single cpu(0~7).
And each cpu event(a bar) has raw data with ts(begin timestamp) and ts_end(end timestamp), and then dur(ts_end-ts)
![image](https://github.com/altair-viz/altair/assets/72813698/a8c4e5ca-39ee-49a0-8d78-7bd6c0b4c14f)
Once I apply zoom operation on the chart above, I will apply an interval selection, which bound to scales x, to the chart below, to get a hist, with sum(dur) of the events selected in the chart above.
![image](https://github.com/altair-viz/altair/assets/72813698/56d267e1-189a-40e2-8dc0-4d028b833dca)

My problem is, when the events was cut by the interval, the hist will be incorrect:
- some events will not be counted, because ts < interval.min()
- some events should be clamped because ts_end > interval.max()
- this opeartion in perfetto(google's performance analysis framework), we call it span join
![image](https://github.com/altair-viz/altair/assets/72813698/181081ae-d3c2-49b8-b856-c73fedcbfc52)

So could I apply span join in the hist chart?

**Reply by mattijn** [reply]
Thanks. I understand the issue you are having. I don't have a solution yet, but a minimal, reproducible example could be something as such:
```python
import altair as alt
import pandas as pd

source = pd.DataFrame([
    {"task": "A", "start": 1, "end": 4},
    {"task": "B", "start": 3, "end": 7},
    {"task": "C", "start": 7, "end": 10}
])

brush = alt.selection_interval(encodings=['x'], empty=False, value={'x': [3.5, 8]})

bar_source = alt.Chart(source, title='source chart').mark_bar().encode(
    x='start',
    x2='end',
    y='task'
).add_params(brush)

bar_filtered = alt.Chart(source, title='filtered chart').mark_bar().encode(
    x='start',
    x2='end',
    y=alt.Y('task').scale(domain=['A','B','C'])
).transform_filter(brush)

bar_source & bar_filtered
```
![image](https://github.com/altair-viz/altair/assets/5186265/da354477-d857-41d2-b399-a95c564dbab3)

Where the requested result would be a filtered bar chart with the exact view of the brush interval selection. Interesting question!
